lib/apiservers/engine/backends/archive.go

```diff
@@ -190,9 +198,18 @@ func (c *Container) ContainerStatPath(name string, path string) (stat *types.Con
 		return nil, NotFoundError(name)
 	}
 
+	path = filepath.Clean(path)
 	mounts := mountsFromContainer(vc)
 	mounts = append(mounts, types.MountPoint{Destination: "/"})
 
+	// handle the special case of targeting a volume mount point before it exists.
+	// this will be important for non started container cp, will also be important
+	// to certain behaviors for diff on a non started container.
+	if stat, succeed := tryFakeStatPath(mounts, path); succeed {
```

Comment:
I'm not clear why we're doing this before we try in the real filesystem. Shouldn't the logic be:
a. if target found via full stat path, return actual stat results
b. if target not found and target is known to be a mount point, return fake mount dir stat.